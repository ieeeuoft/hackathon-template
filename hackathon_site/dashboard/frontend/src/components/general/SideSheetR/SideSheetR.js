import React from "react";
import Drawer from "@material-ui/core/Drawer";
import Typography from "@material-ui/core/Typography";
import Link from "@material-ui/core/Link";
import Button from "@material-ui/core/Button";
import MenuItem from "@material-ui/core/MenuItem";
import FormControl from "@material-ui/core/FormControl";
import Select from "@material-ui/core/Select";
import LaunchIcon from "@material-ui/icons/Launch";

import styles from "./SideSheetR.module.scss";
import { ReactComponent as BackArrow } from "../../../assets/images/icons/BackArrow.svg";
import { ReactComponent as CircleType } from "../../../assets/images/icons/redtype.svg";

const createQuantityList = (number) => {
    var entry = [];

    for (var i = 0; i < number; i++) {
        entry.push(i + 1);
    }

    return entry;
};

const CartSheet = ({ cart, detail }) => {
    const handleChange = (value) => {
        alert(value);
    };

    if (cart) {
        return (
            <div className={styles.cart}>
                <FormControl className={styles.form}>
                    <Select>
                        {createQuantityList(detail.quantity).map((val, i) => (
                            <MenuItem onClick={() => handleChange(val)} key={i}>
                                {val}
                            </MenuItem>
                        ))}
                    </Select>
                </FormControl>
                <Button
                    variant="contained"
                    className={styles.cartButton}
                    onClick={cart}
                    disableElevation
                >
                    ADD TO CART
                </Button>
            </div>
        );
    } else {
        return null;
    }
};

const BodySheet = ({ detail }) => {
    return (
        <div className={styles.bodysheet}>
            <div className={styles.bodyinfo}>
                <Typography variant="h3">Manufacturer</Typography>
                <Typography variant="p">{detail.manufacturer}</Typography>
            </div>
            <div className={styles.bodyinfo}>
                <Typography variant="h3">Model Number</Typography>
                <Typography variant="p">{detail.model_num}</Typography>
            </div>
            <div className={styles.bodyinfo}>
                <Typography variant="h3">Datasheet</Typography>
                <Link
                    href={detail.datasheet}
                    rel="noopener"
                    color="inherit"
                    underline="none"
                    target="_blank"
                    style={{ display: "flex" }}
                >
                    <LaunchIcon></LaunchIcon>
                    <Typography variant="p">Link</Typography>
                </Link>
            </div>
            <div className={styles.bodyinfo}>
                <Typography variant="h3">Notes</Typography>
                {detail.notes.map((note, i) => (
                    <Typography variant="p" key={i}>
                        {note}
                    </Typography>
                ))}
            </div>
            <div className={styles.bodyinfo}>
                <Typography variant="h3">Constraints</Typography>
                {detail.constraints.map((constraint, i) => (
                    <Typography variant="p" key={i}>
                        {constraint}
                    </Typography>
                ))}
            </div>
        </div>
    );
};

const HeaderSheet = ({ detail }) => {
    return (
        <div className={styles.headersheet}>
            <div>
                <div className={styles.title}>
                    <CircleType
                        style={{ color: detail.type, margin: "0 5px" }}
                    ></CircleType>
                    <Typography variant="h2">{detail.name}</Typography>
                </div>
                <Typography className={styles.available}>
                    {detail.available} OF {detail.total} IN STOCK
                </Typography>
                <div className={styles.tags}>
                    <Typography variant="h3">Tags</Typography>
                    <Typography variant="p">{detail.tags}</Typography>
                </div>
            </div>
            <img src={detail.img} alt="product" />
        </div>
    );
};

const SideSheetR = ({ detail, cart }) => {
    const [toggle, setToggle] = React.useState(false);

    const handleChange = (change) => {
        setToggle(change);
    };

    return (
        <div>
            <div
                onClick={() => {
                    handleChange(true);
                }}
            >
                click me
            </div>
            <Drawer anchor="right" open={toggle}>
                <div className={styles.topsheet}>
                    <BackArrow
                        className={styles.backarrow}
                        onClick={() => {
                            handleChange(false);
                        }}
                    />
                    <Typography variant="h2" className={styles.title}>
                        Product Overview
                    </Typography>
                </div>
                <HeaderSheet detail={detail} />
                <BodySheet detail={detail} />
                <CartSheet cart={cart} detail={detail} />
            </Drawer>
        </div>
    );
};

export default SideSheetR;
