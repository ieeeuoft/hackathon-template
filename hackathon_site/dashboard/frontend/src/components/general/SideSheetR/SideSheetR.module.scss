@import "./../../../assets/abstracts/mixins";
@import "./../../../assets/abstracts/variables";

.topsheet {
    background-color: color(grey);
    display: flex;
    padding: 23px 90px;
    width: 100%;
    justify-content: flex-start;
    .backarrow {
        margin: 0 15px;
    }
}
