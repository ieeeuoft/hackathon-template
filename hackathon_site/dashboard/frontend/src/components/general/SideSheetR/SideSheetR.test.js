import React from "react";
import SideSheetR from "./SideSheetR";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

/**
 * Test the Add Cart function
 * Test the quantity log thing
 * Test the back button
 * Test with snapshot
 */

describe("SideSheetR", () => {
    const testDetail = {
        name: "Arduino",
        type: "red",
        total: 30,
        available: 29,
        img: "https://www.filterforge.com/more/help/images/size200.jpg",
        tags: "MCU, FPGA",
        manufacturer: "Canakit",
        model_num: "Model 3B+",
        datasheet: "link",
        notes: ["- For micropython ask for image", "- randomnerdtutorials.com"],
        constraints: ["- Max 1 of this item", "- Max 3 microcontroller labelled red"],
        quantity: 3,
    };

    const addCartMock = jest.fn();

    test("render", () => {
        const { asFragment } = render(
            <SideSheetR detail={testDetail} cart={addCartMock} />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    // I will write more tests after I get feedback for the component
});
