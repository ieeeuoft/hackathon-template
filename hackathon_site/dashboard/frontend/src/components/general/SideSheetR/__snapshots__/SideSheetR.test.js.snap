// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideSheetR render 1`] = `
<DocumentFragment>
  <div>
    <div>
      click me
    </div>
  </div>
</DocumentFragment>
`;
